woltka/ordinal.py

```diff
@@ -320,5 +320,5 @@ def add_match_to_readmap(rmap, match, rids, nucl=None):
     """
     for idx, genes in match.items():
         if nucl:
-            genes = {'{}_{}'.format(nucl, x) for x in genes}
+            genes = {nucl + '_' + x for x in genes}
```

Comment:
Why did you make this change? It seems more brittle to me.

Reply:
I actually did a careful benchmark of multiple scenarios. Finally, and to my surprise, I found that this form is the fastest. :)

Reply:
Even slightly faster than `'_'.join((nucl, x))`. Because this function is one of the bottlenecks of the algorithm, I tried to optimize each step.

Reply:
Okay, can you share the benchmarking results and script in an issue? So we have it for future reference if this issue comes up in the future.

Reply:
Actually, I just did more tests and found that f-string is an even better solution. Issue created (#23).

```
In [1]: nucl = 'nucl'                                                                                                   

In [2]: gene = 'gene'                                                                                                   

In [3]: %timeit '%s_%s' % (nucl, gene)                                                                                  
273 ns ± 3.53 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [4]: %timeit '{}_{}'.format(nucl, gene)                                                                              
366 ns ± 3.33 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [5]: %timeit f'{nucl}_{gene}'                                                                                        
146 ns ± 0.04 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [6]: %timeit '_'.join((nucl, gene))                                                                                  
221 ns ± 0.224 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [7]: %timeit nucl + '_' + gene                                                                                       
192 ns ± 1.89 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
This conclusion also holds in this particular context:

```
In [1]: nucl = 'nucl'  

In [2]: genes = [f'gene{i}' for i in range(100)]                                                                        

In [3]: %timeit {'%s_%s' % (nucl, x) for x in genes}                                                                   
34.5 µs ± 27.6 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [4]: %timeit {'{}_{}'.format(nucl, x) for x in genes}                                                               
43.6 µs ± 68.8 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [5]: %timeit {f'{nucl}_{x}' for x in genes}                                                                          
23.1 µs ± 264 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %timeit {'_'.join((nucl, x)) for x in genes}                                                                   
31.1 µs ± 54.9 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [7]: %timeit {nucl + '_' + x for x in genes}                                                                         
27.1 µs ± 71 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```